src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -671,9 +671,28 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
             $dumper->setProxyDumper(new ProxyDumper(md5($cache->getPath())));
         }
 
-        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'debug' => $this->debug));
+        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'as_files' => !$this->debug, 'debug' => $this->debug));
```

Comment:
The idea is to keep the generated file to a single file in debug mode (so it can be easily looked at and checked) and use the optimizations from this PR (breaking things out into separate files) when in production mode. Not that I agree or disagree with this, just wanted to provide the context you asked.